nntrainer/dataset/batch_queue.cpp

```diff
@@ -68,4 +68,80 @@ bool BatchQueue::isEmpty() const {
   return q.empty();
 }
 
+IterationQueue::IterationQueue(
+  unsigned int num_slots, const std::vector<ml::train::TensorDim> &input_dims,
+  const std::vector<ml::train::TensorDim> &label_dims) {
+  iterations.reserve(num_slots);
+  for (decltype(num_slots) i = 0; i < num_slots; ++i) {
+    iterations.emplace_back(input_dims, label_dims, this);
+    empty_q.push(&iterations.back());
+  }
+}
+
+ScopedView<Sample> IterationQueue::requestEmpty() {
+  if (being_filled == nullptr) {
+    being_filled = empty_q.front();
+    empty_q.pop();
+    current_iterator = being_filled->get().begin();
+  } else {
+    current_iterator++;
+  }
+
+  auto view = ScopedView<Sample>(&(*current_iterator),
+                                 [this] { being_filled->markSampleFilled(); });
```

Comment:
being_filled is taken from this but being_filled might be reset to nullptr. I am not sure if it is safe to use being_filled like this.

Reply:
Yeah, it's unsafe. It is changed in the following PR https://github.com/nnstreamer/nntrainer/pull/1488/commits/f089ae13ca53c04d4c8e882dd5a197a8e75b03ce#diff-45e2e03d99d3d4d24a28897fb30cf19bf900f94742302f6b8420b0d1717dbaf0R99-R101